src/main/java/org/example/ByteValidator.java

```
@@ -1,20 +1,29 @@
 package org.example;
+import java.nio.charset.Charset;
+import java.nio.charset.CharsetDecoder;
+import java.nio.ByteBuffer;
+import java.nio.CharBuffer;
 
 public class ByteValidator {
+    String charsetName;
 
-    public String validCharacters;
-
-    public ByteValidator(String validCharacters) {
-        this.validCharacters = validCharacters;
+    public ByteValidator(String charsetName) {
+        this.charsetName = charsetName;
     }
 
     public boolean isValidByteArray(byte[] byteArray) {
-        for (byte b : byteArray) {
-            char c = (char) b;
-            if (validCharacters.indexOf(c) == -1) {
-                return false;
-            }
+        if (!Charset.isSupported(charsetName)) {
+            return false;
```

Throw exception instead of returning `false` here